src/main/java/org/commcare/core/network/ModernHttpRequester.java

```diff
@@ -208,31 +150,71 @@ public static void processResponse(HttpResponseProcessor responseProcessor,
         }
     }
 
+    /**
+     * Only gets called if response processor is supplied
+     *
+     * @return Input Stream from cache
+     * @throws IOException
+     */
     @Override
     public InputStream getResponseStream() throws IOException {
-        InputStream connectionStream = httpConnection.getInputStream();
+        InputStream inputStream = response.body().byteStream();
+        BitCache cache = BitCacheFactory.getCache(cacheDirSetup, getContentLength(response));
+        cache.initializeCache();
+        OutputStream cacheOut = cache.getCacheStream();
+        StreamsUtil.writeFromInputToOutputNew(inputStream, cacheOut);
+        return cache.retrieveCache();
+    }
 
-        long dataSizeGuess = setContentLengthProps(httpConnection);
-        BitCache cache = BitCacheFactory.getCache(cacheDirSetup, dataSizeGuess);
+    public static RequestBody getPostBody(HashMap<String, String> inputs) {
+        FormBody.Builder formBodyBuilder = new FormBody.Builder();
+        for (Map.Entry<String, String> param : inputs.entrySet()) {
+            formBodyBuilder.add(param.getKey(), param.getValue());
+        }
+        return formBodyBuilder.build();
+    }
 
-        cache.initializeCache();
+    public static String getCredential(User user, String domain) {
```

Comment:
I think these got deleted from this class, but not yet added to `HttpUtils`, so there are compilation issues now